apis/logging/v1/cluster_log_forwarder_types.go

```diff
@@ -147,11 +147,33 @@ type OutputSpec struct {
 	// Secret for authentication.
 	// Name of a secret in the same namespace as the cluster logging operator.
 	//
-	// For client authentication, set secret keys `tls.crt` and `tls.key` to the client certificate and private key.
+	// All sensitive authentication information is provided via a kubernetes Secret object.
+	// A Secret is a key:value map, common keys are described here.
+	// Some outputs support additional pecialized keys, documented with the output-specific configuration field.
+	// All secret keys are optional, enable the security features you want by setting the relevant keys.
 	//
-	// To use your own certificate authority, set secret key `ca-bundle.crt`.
+	// Transport Layer Security (TLS)
 	//
-	// Depending on the `type` there may be other secret keys that have meaning.
+	// Basic TLS is enabled by TLS URL such as 'https://...' or 'ssl://...', it does not require a Secret.
+	// For basic TLS, the client authenticates the server using the system's default certificate authority.
+	//
+	// Additional TLS features are enabled by selectively setting the following optional fields.
+	//
+	//   `tls.crt`: (string) File name of a client certificate to authenticate the client. Enables mutual authentication. Requires `tls.key`.
+	//   `tls.key`: (string) File name containing the private key  to unlock the client certificate. Requires `tls.crt`
+	//   `passphrase`: (string) Passphrase to decode an encoded private key. Requires tls.key.
+	//   `ca-bundle.crt`: (string) File name of a custom CA for server authentication.
+	//
+	// Username and Password
+	//
+	//   `user`: (string) Authentication user name. `password` must also be set.
+	//   `password`: (string) Authentication password. `user` must also be set.
+	//
+	// Simple Authentication Security Layer (SASL)
+	//
+	//   `sasl.enable`: (boolean) Enable or disable SASL.
```

Comment:
Do we need to explicitly "enable" or is it sufficient the presence of something else  means it is enabled?